awsbestpractices/loops-to-batches/LoopsToBatches.java

```diff
@@ -0,0 +1,185 @@
+import org.quartz.Job;
+import org.quartz.JobDataMap;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.quartz.JobKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+
+public class LoopsToBatches {
+    AmazonDynamoDB dynamoDBClient = AmazonDynamoDBClientBuilder.standard().build();
+
+    public void batchablePutItemOnCollection_1(JobExecutionContext context) throws JobExecutionException {
+        MessageConsumer consumer = session.createConsumer(queue);
+        for (int i = 1; i < numMsgs + 1; i++) {
+            Map<String, AttributeValue> item = new HashMap<String, AttributeValue>();
+            TextMessage txtMessage = (TextMessage) consumerMessage;
+            String msgBody = txtMessage.getText().toString();
+            item.put("msg_body", new AttributeValue().withS(msgBody));
+            // Can be batched
+            dynamoDBClient.putItem(getTableName, item);
+        }
+    }
+
+    // Based on https://github.com/locationtech/geowave/blob/5db4f2ba3be6094d6bedfb908afb5126469c76fa/extensions/datastores/dynamodb/src/main/java/org/locationtech/geowave/datastore/dynamodb/operations/DynamoDBWriter.java#L156
+    private void batchablePutItemOnCollection_2(final Map<String, List<WriteRequest>> map) {
+        for (final Entry<String, List<WriteRequest>> requests : map.entrySet()) {
+            for (final WriteRequest r : requests.getValue()) {
+                if (r.getPutRequest() != null) {
+                    dynamoDBClient.putItem(requests.getKey(), r.getPutRequest().getItem());
+                }
+            }
+        }
+    }
+
+    private void notInALoop(String tableName) {
+        // Not in a loop
+        dynamoDBClient.putItem(getTableName, new HashMap<>());
+    }
+
+    // Based on https://github.com/apache/metamodel/blob/3df020a31be3a4c4d82d1305670878b25d85bcef/dynamodb/src/main/java/org/apache/metamodel/dynamodb/DynamoDbDataContext.java#L288
+    public void dynamoWrongGetItem() {
+        final GetItemResult item = dynamoDBClient.getItem(getItemRequest);
+
+        final Object[] values = new Object[selectItems.size()];
+        for (int i = 0; i < values.length; i++) {
+            // Wrong getItem! Must not be flagged
+            final AttributeValue attributeValue = item.getItem().get(attributeNames.get(i));
+            values[i] = DynamoDbUtils.toValue(attributeValue);
+        }
+    }
+
+    // Based on https://github.com/awslabs/aws-codedeploy-plugin/blob/6b74409a2f44faf712ca704ba095cac95fdc5c42/src/main/java/com/amazonaws/codedeploy/AWSCodeDeployPublisher.java#L486
+    private boolean waiter(AWSClients aws, String deploymentId) throws InterruptedException {
+        AmazonCodeDeployClient codeDeployClient = new AmazonCodeDeployClient();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Client constructors are now deprecated in favor of using builders to create the client. Consider using the builder implementation instead.

[Learn more](https://aws.amazon.com/blogs/developer/client-constructors-now-deprecated/)